### ISSUE TITLE
Replace get_mallows_loglik with log_lik_db?

### ISSUE THREAD

**Issue by wleoncio** [ocbe-uio/BayesMallows#91]
`get_mallows_loglik()` is a short function calling `get_rank_distance()` and making a small calculation afterwards. `log_lik_db()` (on `misc_likelihood.R`) is more complex, but supposedly has the basic same purpose. Maybe the former could be replaced with the latter for code simplification.

On the other hand, the former is already on C++ and the latter is in R, which is potentially slower. Either the latter should then be translated, then? Or maybe just drop this idea.

**Comment by osorensen** [comment]
This script shows the relationship between [get_mallows_loglik](https://github.com/ocbe-uio/BayesMallows/blob/master/src/smc_get_mallows_loglik.cpp), [lik_db_mix](https://github.com/ocbe-uio/BayesMallows/blob/master/R/lik_db_mix.R), and [log_lik_db](https://github.com/ocbe-uio/BayesMallows/blob/master/R/misc_likelihood.R). 

``` r
library(BayesMallows)
alpha <- 10
N <- 100
n_items <- 5
rho <- 1:n_items
metric <- "footrule"
mydata <- sample_mallows(
  n_samples = N,
  rho0 = 1:n_items,
  alpha0 = 10,
  metric = "footrule")

pfd <- BayesMallows:::partition_function_data
card <- pfd[pfd$metric == metric & pfd$n_items == n_items & pfd$type == "cardinalities", ][["values"]][[1]]
Z <- BayesMallows:::get_partition_function(n_items, alpha, cardinalities = card)


lik_db_mix(
  rho = rho,
  alpha = alpha,
  weights = 1,
  metric = metric,
  rankings = mydata,
  log = TRUE
)
#> [1] -51.45874
BayesMallows:::log_lik_db(rho, alpha / n_items, metric, mydata, rep(1, nrow(mydata)))
#> [1] -51.45874

get_mallows_loglik(
  alpha = alpha, rho = rho, n_items = n_items, rankings = mydata, metric = metric
) - N * Z
#> [1] -51.45874
```

<sup>Created on 2022-05-04 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by osorensen** [comment]
`get_mallows_loglik` does not include the normalizing constant (partition function). However, this is done a few other places, e.g., here

https://github.com/ocbe-uio/BayesMallows/blob/68707cd151a4d900ec35126559feb9c1257321e5/src/smc_mallows_new_users_partial.cpp#L164-L170

I'll try to merge these, and if necessary have an argument specifying whether to scale by the normalizing constant or not.

**Comment by osorensen** [comment]
@anjastein and @wleoncio, I'd like to have your opinion on this issue before I make a PR.

It seems to me that `get_mallows_loglik` does not compute the log-likelihood of the Mallows model, but rather just the exponent. As you see from the comparisons I made above, the result of `get_mallows_loglik` coincides with the results of `lik_db_mix` if I subtract the number of observations multiplied by the logarithm of the partition function.

My suggestion is that we rename `get_mallows_loglik` to something else, e.g., `get_exponent_sum`, and make it an internal function for the package. To do this would be a simple find-and-replace job. Once this is done, I can start improving the code for `lik_db_mix`, and perhaps eventually rename that function to `get_mallows_loglik`, which I think is a much better name.

Ok?

**Comment by wleoncio** [comment]
Sounds like a solid idea, just gotta go carefully so that the old and the new `get_mallows_loglik()` don't co-exist at any moment.

**Comment by osorensen** [comment]
Closed by #231
